haystack/nodes/search_engine/base.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+from haystack import Document
+
+
+class SearchEngine(ABC):
+    """
+    Abstract base class for search engines providers.
+    """
+
+    @abstractmethod
+    def search(self, query, **kwargs) -> List[Document]:
```

Comment:
query type is not given but we can define it as str or do you expect other formats?